pkg/agent/openflow/feature.go

```diff
@@ -0,0 +1,238 @@
+// +-------------+ +-------------+ +-------------+ +-------------+ +-------------+ +-------------+ +-------------+
+// |   table A   | |   table B   | |   table C   | |   table D   | |   table E   | |   table F   | |   table     |
+// |   stage S1  | |           stage S2          | |           stage S1          | |          stage S4           |
+// |                   pipeline P                | |                          pipeline Q                         |
+// +-------------+ +-------------+ +-------------+ +-------------+ +-------------+ +-------------+ +-------------+
+//          \              |              /                \                          |                  /
+//           \             |             /                  \ ------ \                |          /-----/
+//            \            |            /                             \               |         /
+//            +------------------------+                              +-------------------------+
+//            |       pipeline P       |                              |        pipeline Q       |
+//            |       - table A        |                              |        - table D        |
+//            |       - table B        |                              |        - table F        |
+//            |       - table C        |                              |        - table G        |
+//            +------------------------+                              +-------------------------+
+
+// feature is the interface to program a major function in Antrea data path. The following structures implement this interface:
+// - featurePodConnectivity, active by default, implementation of connectivity for Pods.
+// - featureNetworkPolicy, active by default, implementation of K8s NetworkPolicy and Antrea NetworkPolicy.
+// - featureService, active by default, implementation of K8s Service.
+// - featureEgress, activation is determined by feature gate Egress, implementation of Egress.
+// - featureMulticast, activation is determined by feature gate Multicast, implementation of multicast.
+// - featureTraceflow, implementation of Traceflow.
+type feature interface {
+	// getFeatureName returns the name of the feature.
+	getFeatureName() string
+	// getRequiredTables returns a slice of required tables of the feature.
+	getRequiredTables() []*Table
+	// initFlows returns the initial flows of the feature.
+	initFlows() []binding.Flow
+	// replayFlows returns the fixed and cached flows that need to be replayed after OVS is reconnected.
+	replayFlows() []binding.Flow
+}
+
+// Table in FlexiblePipeline is the basic unit to build OVS pipelines. A Table can be referenced by one or more features,
+// but its member struct ofTable will be initialized and realized on OVS only when it is referenced by any active features.
+// Note that, default active features are PodConnectivity, NetworkPolicy and Service. The activation of feature Egress
```

Comment:
I feel we can remove this, as it just repeats what are described in the overview section earlier.

Reply:
Removed.